compiler/src/iree/compiler/Codegen/LLVMGPU/LLVMGPUVectorLowering.cpp

```diff
@@ -183,6 +184,52 @@ struct SetMulAddFMF final : OpRewritePattern<vector::MultiDimReductionOp> {
   }
 };
 
+struct UnrollElementwiseOps : public RewritePattern {
+  UnrollElementwiseOps(MLIRContext *context, PatternBenefit benefit = 1)
+      : RewritePattern(MatchAnyOpTypeTag(), benefit, context) {}
```

Comment:
SRIRV uses upstream unrolling https://github.com/iree-org/iree/blob/08efffa523e13a420b5f8548e382c839a199b371/compiler/src/iree/compiler/Codegen/SPIRV/SPIRVInitialVectorLowering.cpp#L257

Are you not using this because of the strided slices it uses?

Reply:
Long term, we should use the same unrolling implementation for both llvm and spirv and then have a secondary phase where we break down long vectors for spirv